docs: upload tute for setting up sql review ci (#675)
bytebase/bytebase.com: content/blog/sql-review-ci-tute.md

---

> title: Integrating SQL Review into Your GitLab & GitHub CI/CD

Reviewer: how-to-integrate-sql-review-into-gitlab-github-ci. (because that's more SEO friendly)

How to integrate SQL Review into Your GitLab / GitHub CI

Remove CD part from all other places as well.

Reviewer: GitLab/GitHub only plays the CI role here. CD means continuous deployment which is a different thing from CI (Continuous Integration).

Reviewer 2: But CI/CD is better for SEO?

Reviewer: I see

Author: I've renamed the title to "How to integrate SQL Review into Your GitLab or GitHub CI/CD", tag is "how-to-integrate-sql-review-into-gitlab-github-ci"

author: Mila
published_at: 2022/12/2 21:21:21
feature_image: /static/blog/sql-review-ci-tute/sql-review-ci.webp
tags: How-To
description: This is a tutorial on how to integrate SQL Review into your GitLab & GitHub CI/CD, so that SQL review is automatically triggered before SQL scripts are merged into your repos.
---

> To accommodate to the different ways different development teams work, Bytebase offers multiple ways to integrate SQL Review into your workflow.

Reviewer: ```suggestion
Different development teams adopt different development workflow, and Bytebase offers multiple ways to integrate SQL Review into their respective workflows.
```

This article will walk you through how to seamlessly integrate SQL review into your GitHub and/or GitLab CI/CD workflow, so that SQL review is automatically triggered before SQL scripts are merged into your repos.

For the sake of this tutorial, we will use GitLab as an example, GitHub works similarly.

## Prerequisites

> This feature requires the team plan. (Starting from v1.8.0, you can start a 14-day team plan trial without registration!)

Reviewer: It's enterprise plan only

Please enable the [VCS workflow](/docs/vcs-integration/overview) for your Bytebase workspace and project.

> ## Step 1. Configure SQL review CI

Reviewer 2: Step 1 - Configure SQL review CI

After you enable VCS workflow from the Bytebase Console, under **Version Control** in your project, check the box to enable **SQL Review CI**.

![_](/static/blog/sql-review-ci-tute/enable-sql-review-ci.webp)

Click **Update.** Bytebase will create a Merge Request (MR) in your GitLab repository to set up SQL review CI.

![_](/static/blog/sql-review-ci-tute/setup-sql-review-ci.webp)

You will be redirected to your GitLab repository and see that an MR has been generated. Please review and merge this MR to complete the setup.

![_](/static/blog/sql-review-ci-tute/gitlab-mr.webp)

## Step 2. SQL Review via CI

Once you've set up the CI, the SQL review policy will be automatically triggered to check against each MR that contains SQL script files.

![_](/static/blog/sql-review-ci-tute/gitlab-sql-review.webp)

The following image shows the detailed results of a SQL review.

> ![_](/static/blog/sql-review-ci-tute/gha-sql-review-details.webp)

Reviewer: This is a GitHub screenshot.

Author: moved the screenshot down a section :)

Similarly, in GitHub repos, SQL review will check for changes that match [the file path template](/docs/vcs-integration/name-and-organize-schema-files#file-path-template) through GitHub Actions.

Follow [this doc](/docs/sql-review/review-rules/create-schema-review-policy) to create and configure your own SQL review policy.

## Summary

Bytebase now supports integrating SQL review CI into your preferred code repository, whether it be GitHub or GitLab. In addition, for GitHub, Bytebase provides [additional ways](/docs/sql-review/sql-advisor/overview) through:

> - GitHub App: 0 codings required.

Reviewer: ```suggestion
- [GitHub App](https://github.com/marketplace/bytebase): Zero code configuration.
```
